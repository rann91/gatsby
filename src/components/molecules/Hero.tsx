/* @jsx jsx */
import Img, { FluidObject } from 'gatsby-image'
import { Heading, jsx, Text, Container, Box } from 'theme-ui'
import { rem } from '../../gatsby-plugin-theme-ui'
import { Strapi_ComponentContentHero } from '../../typings/graphql'
import Link from '../atoms/Link'
import { Fragment } from 'react'

const Hero = ({
  title,
  description,
  image,
  link
}: Strapi_ComponentContentHero) => {
  return (
    <Box
      as={title ? 'section' : 'div'}
      sx={{
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        minHeight:
          title && description && link
            ? [rem(480), null, null, rem(620)]
            : [rem(320), null, null, rem(480)],
        backgroundColor: 'black'
      }}>
      {image && (
        <Fragment>
          <Img
            style={{
              position: 'absolute',
              top: 0,
              left: 0,
              width: '100%',
              height: '100%'
            }}
            fluid={
              (image!.file!.childImageSharp!.fluid as FluidObject) || undefined
            }
          />
          <div
            sx={{
              position: 'absolute',
              top: 0,
              left: 0,
              width: '100%',
              height: '100%',
              backgroundColor: 'rgba(0, 0, 0, 0.4)',
              pointerEvents: 'none'
            }}
          />
        </Fragment>
      )}
      {(title || description || link) && (
        <Container
          px={5}
          sx={{ position: 'relative', color: 'white', textAlign: 'center' }}>
          {title && (
            <Heading as="h1" variant="jumbo">
              {title}
            </Heading>
          )}
          {description && (
            <Text
              as="p"
              sx={{
                '* + &': {
                  paddingTop: 2
                }
              }}>
              {description}
            </Text>
          )}
          {link && (
            <Link
              link={link}
              sx={{
                variant: 'buttons.lightLarge',
                '* + &': {
                  marginTop: [5, null, null, 6]
                }
              }}
            />
          )}
        </Container>
      )}
    </Box>
  )
}

export default Hero
