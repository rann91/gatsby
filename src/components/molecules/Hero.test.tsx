import React from 'react'
import data from '../../json/hero.json'
import { render } from '@testing-library/react'
import Hero from './Hero'

describe('Hero', () => {
  it('renders correctly', () => {
    const { container } = render(<Hero {...data} />)

    expect(container).toMatchSnapshot()
  })
})
