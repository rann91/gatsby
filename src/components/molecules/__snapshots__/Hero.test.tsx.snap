// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<div>
  <section
    class="css-cwo03m-Hero"
    data-testid="section"
  >
    <div
      class=" gatsby-image-wrapper"
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%; height: 100%;"
    >
      <div
        aria-hidden="true"
        style="width: 100%; padding-bottom: 66.66666666666667%;"
      />
      <img
        alt=""
        aria-hidden="true"
        src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAANABQDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAABAAC/8QAFQEBAQAAAAAAAAAAAAAAAAAAAwT/2gAMAwEAAhADEAAAAT5eOdx2JT//xAAZEAACAwEAAAAAAAAAAAAAAAAAAQIREiH/2gAIAQEAAQUCkldIaV4wT4OXf//EABcRAAMBAAAAAAAAAAAAAAAAAAECEEH/2gAIAQMBAT8BYbP/xAAWEQEBAQAAAAAAAAAAAAAAAAACEBL/2gAIAQIBAT8BD1P/xAAYEAACAwAAAAAAAAAAAAAAAAAQESAxgf/aAAgBAQAGPwKlDR//xAAbEAEAAgIDAAAAAAAAAAAAAAABABEQQSFRYf/aAAgBAQABPyG/bdvI4cAmm7dx8w0i40f/2gAMAwEAAgADAAAAEHPv/8QAFhEBAQEAAAAAAAAAAAAAAAAAABEh/9oACAEDAQE/EIGP/8QAFxEBAAMAAAAAAAAAAAAAAAAAAAEhMf/aAAgBAgEBPxCDC3//xAAbEAEAAwEAAwAAAAAAAAAAAAABABEhMUFhof/aAAgBAQABPxDZ6AFDVeo0ppteE4xLbiYm+wwHdx3e/IleJ//Z"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
      />
      <picture>
        <source
          sizes="(max-width: 1920px) 100vw, 1920px"
          srcset="/static/b81e707d3f2049269e1ccb8818c45960/9104c/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 480w,
/static/b81e707d3f2049269e1ccb8818c45960/a6352/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 960w,
/static/b81e707d3f2049269e1ccb8818c45960/d8255/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 1920w,
/static/b81e707d3f2049269e1ccb8818c45960/af56c/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 2880w,
/static/b81e707d3f2049269e1ccb8818c45960/27fce/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 3840w,
/static/b81e707d3f2049269e1ccb8818c45960/1c878/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 4752w"
        />
        <img
          alt=""
          loading="lazy"
          sizes="(max-width: 1920px) 100vw, 1920px"
          src="/static/b81e707d3f2049269e1ccb8818c45960/d8255/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg"
          srcset="/static/b81e707d3f2049269e1ccb8818c45960/9104c/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 480w,
/static/b81e707d3f2049269e1ccb8818c45960/a6352/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 960w,
/static/b81e707d3f2049269e1ccb8818c45960/d8255/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 1920w,
/static/b81e707d3f2049269e1ccb8818c45960/af56c/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 2880w,
/static/b81e707d3f2049269e1ccb8818c45960/27fce/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 3840w,
/static/b81e707d3f2049269e1ccb8818c45960/1c878/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 4752w"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        />
      </picture>
      <noscript>
        &lt;picture&gt;&lt;source srcset="/static/b81e707d3f2049269e1ccb8818c45960/9104c/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 480w,
/static/b81e707d3f2049269e1ccb8818c45960/a6352/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 960w,
/static/b81e707d3f2049269e1ccb8818c45960/d8255/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 1920w,
/static/b81e707d3f2049269e1ccb8818c45960/af56c/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 2880w,
/static/b81e707d3f2049269e1ccb8818c45960/27fce/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 3840w,
/static/b81e707d3f2049269e1ccb8818c45960/1c878/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 4752w" sizes="(max-width: 1920px) 100vw, 1920px" /&gt;&lt;img loading="lazy" sizes="(max-width: 1920px) 100vw, 1920px" srcset="/static/b81e707d3f2049269e1ccb8818c45960/9104c/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 480w,
/static/b81e707d3f2049269e1ccb8818c45960/a6352/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 960w,
/static/b81e707d3f2049269e1ccb8818c45960/d8255/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 1920w,
/static/b81e707d3f2049269e1ccb8818c45960/af56c/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 2880w,
/static/b81e707d3f2049269e1ccb8818c45960/27fce/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 3840w,
/static/b81e707d3f2049269e1ccb8818c45960/1c878/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg 4752w" src="/static/b81e707d3f2049269e1ccb8818c45960/d8255/branch_plant_wood_leaf_flower_petal_694258_pxhere_com_711d93fc90.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
      </noscript>
    </div>
    <div
      class="css-1ab51y2-Hero"
    />
    <div
      class="css-71cnkm-Hero"
    >
      <h1
        class="css-1wf7gym"
      >
        Grow your business
      </h1>
      <p
        class="css-1uqf9c8-Hero"
      >
        Give your business a boost with a beautifully crafted homepage.
      </p>
      <a
        class="css-1i3epbt-Hero"
        data-testid="theme-link"
        href="https://www.gatsbyjs.org"
        rel="noopener noreferrer"
        target="_self"
      >
        Learn more
      </a>
    </div>
  </section>
</div>
`;
